1.2.0 - executions pause and resume

igcherkaev opened the pull request:
## 1.2.0

- Added `st2.executions.pause` and `st2.executions.resume` actions.

igcherkaev committed:
Merge pull request #2 from StackStorm-Exchange/master

Fast forwarding to the base repo

igcherkaev committed:
## 1.2.0

- Added `st2.executions.pause` and `st2.executions.resume` actions.

igcherkaev committed:
## 1.2.0

- Added `st2.executions.pause` and `st2.executions.resume` actions.

igcherkaev committed:
## 1.2.0

- Added `st2.executions.pause` and `st2.executions.resume` actions.

igcherkaev committed:
Merge pull request #3 from copartit/dev

1.2.0

nmaludy commented on actions/executions_pause.py:
```diff
@@ -0,0 +1,25 @@
+            try:
+                res = self.client.liveactions.pause(execution_id=i)
+            except Exception as exc:
+                result[i] = '{}'.format(exc)
```
```suggestion
                result[i] = str(exc)
```

emptywee replied:
You got it!

nmaludy commented on actions/executions_pause.py:
```diff
@@ -0,0 +1,25 @@
+                res = self.client.liveactions.pause(execution_id=i)
+            except Exception as exc:
+                result[i] = '{}'.format(exc)
+            else:
```
Also, if one of these fails (exception) then we should probably stop and fail the action, right now the action will always succeed.

emptywee replied:
yes, it's coming! Realized that after running some real-world scenarios :)

emptywee replied:
but not sure if we need to stop? I thought of returning overall result True/False and a dict of results. Perhaps, we can add a flag to ignore/stop on errors as action's parameter? Maybe in a future release if such feature is requested?

nmaludy commented on actions/executions_resume.py:
```diff
@@ -0,0 +1,25 @@
+]
+
+
+def format_result(item):
```
this function looks identical to the one in the file above, it should go into the base action so code is not copy/pasted.

emptywee replied:
:) agreed

nmaludy commented on actions/executions_resume.py:
```diff
@@ -0,0 +1,25 @@
+
+
+def format_result(item):
+    if not item:
+        return None
+
+    return item.to_dict()
```
```suggestion
    return item.to_dict() if item else None
```

emptywee replied:
ok!

igcherkaev committed:
## 1.2.0

- Added `st2.executions.pause`, `st2.executions.resume`, `st2.executions.cancel` actions.

igcherkaev committed:
## 1.2.0

- Added `st2.executions.pause`, `st2.executions.resume`, `st2.executions.cancel` actions.

igcherkaev committed:
## 1.2.0

- Added `st2.executions.pause`, `st2.executions.resume`, `st2.executions.cancel` actions.

emptywee commented:
Yup, and I'll add one more action - `executions.cancel` to have full hand of actions to manage executions :)

igcherkaev committed:
## 1.2.0

- Added `st2.executions.pause`, `st2.executions.resume`, `st2.executions.cancel` actions.

igcherkaev committed:
Merge pull request #4 from copartit/dev

1.2.0 update

igcherkaev committed:
## 1.2.0

- Chatops Aliases for `st2.executions.pause`, `st2.executions.resume` and `st2.executions.cancel` actions.

igcherkaev committed:
## 1.2.0

- Chatops Aliases for `st2.executions.pause`, `st2.executions.resume` and `st2.executions.cancel` actions.

igcherkaev committed:
## 1.2.0

- Chatops Aliases for `st2.executions.pause`, `st2.executions.resume` and `st2.executions.cancel` actions.

igcherkaev committed:
## 1.2.0

- Chatops Aliases for `st2.executions.pause`, `st2.executions.resume` and `st2.executions.cancel` actions.

igcherkaev committed:
## 1.2.0

- Chatops Aliases for `st2.executions.pause`, `st2.executions.resume` and `st2.executions.cancel` actions.

igcherkaev committed:
Merge pull request #5 from copartit/dev

1.2.0 - updated chatops aliases and small bugfix for cancel execution

nmaludy commented on actions/executions_cancel.py:
```diff
@@ -0,0 +1,22 @@
+        for i in ids:
+            try:
+                res = self.client.executions.delete_by_id(instance_id=i)
+                if 'faultstring' in res and res['faultstring']:
```
This could be simplified to 
```suggestion
                if res.get('faultstring'):
```

emptywee replied:
fixed, in the upcoming PR

igcherkaev committed:
code style

igcherkaev committed:
(dummy) updated pack.yaml for release automation

igcherkaev committed:
Merge pull request #6 from copartit/dev

dummy update pack.yaml
